### ISSUE TITLE
A11y: Missing descriptive text in label elements for checkbox

### ISSUE THREAD

**Issue by icemanlily** [IBM/carbon-components-react#1477]
===== DEFECT DESCRIPTION =====
The label element is used to explicitly associate a form control with a label. A label is attached to a specific form control through the use of the "for" attribute. Screen readers use the text in the label as a description of the attached form control.
IBM CI 162 Checkpoint Violation: 4.1.2 Name, Role, Value
IBM Guidance: https://www.ibm.com/able/guidelines/ci162/name_role_value.html

===== EXPECTED RESULT =====
https://aat.w3ibm.mybluemix.net/token/a80a8d09-7f5a-480a-adcc-7e22f77482b0/c19cc7e4-e2e6-4475-89d7-533c2bc34979/archives/2018AugustDeploy//doc/w3/help/en-US/idhi_accessibility_check_g1055.html

![image](https://user-images.githubusercontent.com/6970471/47478913-e1674600-d85d-11e8-93ce-2f4d9897ea89.png)

**Comment by joshblack** [comment]
Hi @icemanlily! 👋 Thanks so much for taking the time to make this issue 😄 

One thing that would help us out a ton is taking a second to fill out the issue template provided for bugs. Would really be helpful to understand what version of carbon components you are using, what component, etc. Thanks!

**Comment by icemanlily** [comment]
The checkbox has hideLabel attribute. It should be fixed this issue. So close it. Thank you!
